### PR TITLE
General performance and memory usage improvements

### PR DESCRIPTION
**Pull request by ammarasyad**
Continuing on the last PR by Reco1l, I decided to incorporate some of his changes to the current osu!droid codebase, with some of my own changes. As this will bring a large change, this will require extensive testing to ensure stability. Report any performance regressions, bugs, or unexpected changes in gameplay.

This PR is mostly a work-in-progress, as I iron out bugs, edge cases, and potential compatibility issues that may arise from the changes. In the meantime, if you wish to help and is able to build osu!droid, please test from my branch. Any input is appreciated.

### CODE REVIEW

File: src/com/edlplan/ui/fragment/InGameSettingMenu.kt [hunk 1]
```diff
@@ -0,0 +1,566 @@
...
+    }
+
+    override fun dismiss() {
+        super.dismiss()
+        uiThread { super.save() }
+        ModMenu.getInstance().hideByFrag()
+    }
+
+    private fun isSettingPanelShow(): Boolean {
+        return abs(findViewById<LinearLayout>(R.id.fullLayout)?.translationY ?: 0f) < 10
```
Comment (Rian8337) [inline src/com/edlplan/ui/fragment/InGameSettingMenu.kt:490]: There is a change in behavior here. Previously, if `findViewById` returns `null`, this would return `false`. Now, it returns `true`. Is this intended?

**Review by Rian8337 — COMMENTED** [review]
Mostly looks good. I have not tested multiplayer yet, since there appears to be multiplayer changes. However, I'm not sure if the change towards speed modify bar (having to check the box first before dragging the slider) is intended, but it brings parity with force difficulty statistics, so I will leave it be.

File: src/com/edlplan/ui/fragment/InGameSettingMenu.kt (hunk 1 above)
Comment (ammarasyad) [reply src/com/edlplan/ui/fragment/InGameSettingMenu.kt:490]: Oops, an oversight in my part. I'll change this accordingly.

**Comment by ammarasyad** [comment]
> Mostly looks good. I have not tested multiplayer yet, since there appears to be multiplayer changes. However, I'm not sure if the change towards speed modify bar (having to check the box first before dragging the slider) is intended, but it brings parity with force difficulty statistics, so I will leave it be.

It is intended. I didn't find the current behavior in release intuitive, so I changed it myself.
